src/librbd/cache/pwl/AbstractWriteLog.cc

```diff
@@ -335,7 +335,7 @@ void AbstractWriteLog<I>::arm_periodic_stats() {
 }
 
 template <typename I>
-void AbstractWriteLog<I>::update_entries(std::shared_ptr<GenericLogEntry> log_entry,
```

Comment:
Why add a reference for a shared pointer?

Reply:
In the original code, the argument i.e. shared_ptr(log_entry) is still null after calling update_entries. It definitely triggers errors when dereferencing the shared_ptr.